src/Security/Security/Cmdlets/SqlVulnerabilityAssessment/AddSqlVulnerabilityAssessmentBaselineRule.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Commands.Security;
+using Microsoft.Azure.Commands.Security.Common;
+using Microsoft.Azure.Commands.SecurityCenter.Models.SqlVulnerabilityAssessment;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.Commands.SecurityCenter.Cmdlets.SqlVulnerabilityAssessment
+{
+    [Cmdlet(VerbsCommon.Add, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlVulnerabilityAssessmentBaselineRule"),OutputType(typeof(PSSqlVulnerabilityAssessmentBaselineResults))]
+    public class AddSqlVulnerabilityAssessmentBaselineRule : SqlVulnerabilityAssessmentSetBaselineBase
+    {
+        [Parameter(ParameterSetName = ParameterSetNames.ResourceIdWithBaselineObject, Mandatory = true, HelpMessage = ParameterHelpMessages.RuleId)]
+        [Parameter(ParameterSetName = ParameterSetNames.OnPremMachinesWithBaselineObject, Mandatory = true, HelpMessage = ParameterHelpMessages.RuleId)]
+        public string RuleId { get; set; }
+
+        [Parameter(ParameterSetName = ParameterSetNames.ResourceIdWithBaselineObject, Mandatory = false, HelpMessage = ParameterHelpMessages.Baseline)]
+        [Parameter(ParameterSetName = ParameterSetNames.OnPremMachinesWithBaselineObject, Mandatory = false, HelpMessage = ParameterHelpMessages.Baseline)]
+        public string[][] Baseline { get; set; }
```

Comment:
The type of Baseline should be HashTable

Reply:
the Add baseline cmdlet works on 1 rule level unlike the Set baseline cmdlet
in the Add baseline scenario the rule id  determines which rule to set the baseline for and the baseline itself is passed as is to the API. using string[][] is a simpler syntax to the user to use. and these parameter has been reviewed here: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/834
Adding a BaselineSet parameter as in the Set baseline cmdlet can be supported in the future in case there's demand for it.

Reply:
As your design doc said, parameter baseline should be hashtable.

```powershell
ResourceIdWithBaselineObject
New-AzSqlVulnerabilityAssessmentBaseline -Baseline <Hashtable> -ResourceId <String> -WorkspaceId <String>
 -Server <String> -Database <String> [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm]
 [<CommonParameters>]
OnPremMachinesWithBaselineObject
New-AzSqlVulnerabilityAssessmentBaseline -Baseline <Hashtable> -WorkspaceId <String> -Server <String>
 -Database <String> -ComputerName <String> -VMUUID <String> -AgentId <String> -WorkspaceResourceId <String>
 [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
```
I think it's little bit of confusion for cutstomer that Baseline in Add-AzSqlVulnerabilityAssessmentBaseline is hashtable but Baseline in Set-AzSqlVulnerabilityAssessmentBaseline is string[][]. what's your thought?
```powershell
ResourceIdWithBaselineLatest
Update-AzSqlVulnerabilityAssessmentBaselineRule -RuleId <String> -ResourceId <String> -WorkspaceId <String>
 -Server <String> -Database <String> [-UseLatest] [-DefaultProfile <IAzureContextContainer>]
 [<CommonParameters>]
OnPremMachinesWithBaselineObject
Update-AzSqlVulnerabilityAssessmentBaselineRule -RuleId <String> -Baseline <String[][]> -WorkspaceId <String>
 -Server <String> -Database <String> -ComputerName <String> -VMUUID <String> -AgentId <String>
 -WorkspaceResourceId <String> [-DefaultProfile <IAzureContextContainer>] [<CommonParameters>]
```

Reply:
At the time I took it with the reviewer online and he asked update the design pr with the decisions we made, but i made a mistake with my update when i wrote "the Add-AzSqlVulnerabilityAssessmentBaselineRule has a -BaselineSet parameter" this was meant for the "Set-AzSqlVulnerabilityAssessmentBaseline"

Set-AzSqlVulnerabilityAssessmentBaseline takes a hash since it updates all the baseline for all the rules all together, thus meaning which means if u had an old baseline object for rule 'A' and the hash table doesn't include a baseline for rule 'A', then the old baseline for rule 'A' will be discarded.
that's why the parameter name in this cmdlet is 'BaselineSet' and not 'Baseline", which is a hashtable. 

while Add-AzSqlVulnerabilityAssessmentBaseline works on a specific rule, it does not discard old baseline that was set before executing the operation and it simply adds to the already existing baseline. that parameter name here in this case is 'Baseline', which is a string[][].

I updated the documentation and added as many examples as possible.

Reply:
Make sense. Thanks for your explanation.

Reply:
Could you update the design doc to facilitate reference?